src/DotNetWorker/Broker/FunctionBroker.cs

```diff
@@ -47,54 +53,70 @@ public async Task<InvocationResponse> InvokeAsync(FunctionInvocation invocation)
 
-                var parameterBindings = executionContext.OutputBindings;
-                var result = executionContext.InvocationResult;
+                    await _functionExecutionDelegate(executionContext);
 
-                foreach (var paramBinding in parameterBindings)
-                {
-                    // TODO: ParameterBinding shouldn't happen here
+                    var parameterBindings = executionContext.OutputBindings;
+                    var result = executionContext.InvocationResult;
 
-                    foreach (var binding in executionContext.OutputBindings)
+                    foreach (var paramBinding in parameterBindings)
                     {
-                        var parameterBinding = new ParameterBinding
+                        // TODO: ParameterBinding shouldn't happen here
+
+                        foreach (var binding in executionContext.OutputBindings)
                         {
-                            Name = binding.Key,
-                            Data = binding.Value.ToRpc()
-                        };
-                        response.OutputData.Add(parameterBinding);
+                            var parameterBinding = new ParameterBinding
+                            {
+                                Name = binding.Key,
+                                Data = binding.Value.ToRpc()
+                            };
+                            response.OutputData.Add(parameterBinding);
+                        }
+                    }
+                    if (result != null)
+                    {
+                        var returnVal = result.ToRpc();
+
+                        response.ReturnValue = returnVal;
                     }
+
+                    response.Result = new StatusResult { Status = Status.Success };
                 }
-                if (result != null)
+                catch (Exception ex)
                 {
-                    var returnVal = result.ToRpc();
-
-                    response.ReturnValue = returnVal;
+                    response.Result = new StatusResult
+                    {
+                        Exception = ex.ToRpcException(),
+                        Status = Status.Failure
+                    };
                 }
-
-                response.Result = new StatusResult { Status = Status.Success };
-            }
-            catch (Exception ex)
-            {
-                response.Result = new StatusResult
+                finally
                 {
-                    Exception = ex.ToRpcException(),
-                    Status = Status.Failure
-                };
-            }
-            finally
-            {
-                (executionContext as IDisposable)?.Dispose();
+                    (executionContext as IDisposable)?.Dispose();
+                }
             }
 
             return response;
         }
-    }
 
+        private static class Log
+        {
```

Comment:
Should we consider moving this to a partial with `<file>.Log.cs` like in the host?(https://github.com/Azure/azure-functions-host/blob/13a43aaa7d7f8441a51233083e692cc584f49143/src/WebJobs.Script.WebHost/Middleware/HostAvailabilityCheckMiddleware.Log.cs)